src/main/java/org/Olivia/EventDispatcher.java

```diff
@@ -0,0 +1,221 @@
+import java.io.File;
+import java.security.InvalidParameterException;
+import java.util.List;
+
+
+public class EventDispatcher {
+
+    private Calendar table;
+    private UiHandler ui;
+    private FileHandler disk;
+
+    public EventDispatcher(Calendar table, UiHandler ui, FileHandler disk) {
+        this.table=table;
+        this.ui=ui;
+        this.disk=disk;
+    }
+
+    private int help() {
+        ui.cout(UiHandler.generateSection(UiHandler.generateHelpMsg()));
+        return 200;
+    }
+    private int list() {
+        ui.cout(UiHandler.generateSection(this.table.toString()));
+        return 200;
+    }
+
+    @Deprecated
+    private int add(String[] entry_info) throws Exception {
+        CalendarEntry entry=new CalendarEntry(entry_info[0]);
+        this.table.addEntry(entry);
+        ui.cout(UiHandler.generateSection("Added: "+entry.toString()+"\n"));
+        disk.syncToFile(this.table);
+        return 200;
+    }
+
+    private int markAsDoneUndone(String input) throws Exception {
+        String[] args=input.toLowerCase().split(" ");
+        if (args.length!=2) {
+            throw new InvalidParameterException("Sorry, which entry do you want me to mark/unmark?");
+            //return 400;
+        }
+        if (args[0].equals("mark")){
+            int status=this.table.markAsDone(Integer.parseInt(args[1]));
+            if (status==200 || status==208) {
+                ui.cout(UiHandler.generateSection("Jawohl, I have marked the following event as completed:\n     " +
+                        this.table.getEntry(Integer.parseInt(args[1]))+"\n"));
+            }
+            disk.syncToFile(this.table);
+            return status;
+        }
+        if (args[0].equals("unmark")){
+            int status=this.table.markAsUndone(Integer.parseInt(args[1]));
+            if (status==200 || status==208) {
+                ui.cout(UiHandler.generateSection("Jawohl, I have marked the following event as incomplete:\n     " +
+                        this.table.getEntry(Integer.parseInt(args[1]))+"\n"));
+            }
+            disk.syncToFile(this.table);
+            return status;
+        }
+        else{
+            throw new RuntimeException("Sorry, there seems to be some difficulties processing your command, could you check the syntax and try again later?");
+            //return 500;
+        }
+    }
+
+    private int delete(String input) throws Exception{
+        String[] args=input.toLowerCase().split(" ");
+        if (args.length!=2){
+            throw new InvalidParameterException("Sorry, which entry do you want me to delete?");
+            //return 400;
+        }
+        ui.cout(UiHandler.generateSection("Jawohl, I have removed the following entry from your calendar:\n     " +
+                this.table.deleteEntry(Integer.parseInt(args[1]))+"\n"));
+        disk.syncToFile(this.table);
+        return 200;
+    }
+
+    private int addEntryToCalendar(String input) throws Exception {
```

Comment:
Might want to consider handing this off to a Parser class or something equivalent! Actually, this can be said for the rest of the code as I notice you often perform string parsing. Perhaps a Parser class with static helper functions may be useful in making your code more readable?

Reply:
Actually the `EventDispatcher` class is designed as the parser. The input is first examined by the `dispatchCommand` function, then parsed and handled by the corresponding functions within the class.